libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.h

```diff
@@ -289,19 +289,40 @@ NS_SWIFT_NAME(SalesforceManager)
 - (nonnull NSString *)devInfoTitleString;
 
 /**
- * Returns the ScreenLockManager instance.
+ * Returns The ScreenLockManager instance.
  *
- * @return the Screen Lock Manager
+ * @return The Screen Lock Manager.
  */
 - (id <SFScreenLockManager>)screenLockManager;
 
 /**
- * Returns the BiometricAuthenticationManager instance.
+ * Returns The BiometricAuthenticationManager instance.
  *
- * @return the Biometric Authentication Manager
+ * @return The Biometric Authentication Manager.
  */
 - (id <SFBiometricAuthenticationManager>)biometricAuthenticationManager;
 
+/**
+ * Creates the NativeLoginManager instance.
+ *
+ * @param remoteAccessConsumerKey The Conneted App consumer key.
+ * @param oauthRedirectURI the Conneted App redirect URI.
+ * @param loginUrl the login url for native login
+ * @param nativeLoginViewController the UI instance that will be presented to the user and
+ *  is responsible for using the returned Native Login Manager to initiate the login process.
+ *
+ * @return The Native Login Manager.
+ */
+- (id <SFNativeLoginManager>)useNativeLogin:(nonnull NSString *)remoteAccessConsumerKey :(nonnull NSString *)oauthRedirectURI
+                                           :(nonnull NSString *)loginUrl :(nonnull UIViewController *)nativeLoginViewController;
```

Comment:
Is this the approach we want going forward? Or do we want to make a new `bootconfig` equivalent?  The main thing that is new here is that each connected app config is tied to and only works with one login url/org.  So we would still need a function like this to pass in the view controller and a name for which connected app to use.

Reply:
No preference on my end! If they fallback to the web view, does the web view use this too or the bootconfig?

small question, is there a reason the parameters aren't named? It looks like the auto complete on the objective-c side doesn't have any other context besides the type

Reply:
Falling back to webview uses the the values from the existing bootconfig for a few reasons but most notably the callback for the new connected app needs to be **a real url** for the org to do the token exchange (and we have seen apps have issues with that).  

RE named parameters:  I'll add it.